docs/source/gateway.rst

```diff
@@ -0,0 +1,745 @@
+.. _gateway:
+
+=============================
+MLflow Gateway (Experimental)
+=============================
+
+.. warning::
+
+    The MLflow Gateway is a new, **experimental feature**. It is subject to modification, feature improvements, or feature removal without advance notice.
+
+The `MLflow Gateway` service is a powerful tool designed to streamline the usage and management of
+various language model providers, such as `OpenAI` and `Anthropic`, within an organization.
+It offers a high-level interface that simplifies the interaction with these services by providing
+a unified endpoint to handle all language model requests.
+
+A major advantage of using the `MLflow Gateway` service is its centralized management of API keys.
+By storing these keys in one secure location, organizations can significantly enhance their
+security posture by minimizing the exposure of sensitive API keys throughout the system. It also
+helps to prevent exposing these keys within code or having end-users have to manage keys safely.
+
+The gateway is designed to be flexible and adaptable, capable of hot-swapping routes if the
+configuration file at the target location is overwritten. This enables the easy incorporation
```

Comment:
I think we should avoid the "hot-swapping" call-out, since we don't want to encourage users to mutate routes that are used in production. On Databricks, we actively prevent users from doing this. We should say that users can easily add and remove routes by updating the configuration file.